### ISSUE TITLE
Overwriting of node file on "bootstrap"

### ISSUE THREAD

**Issue by rmoriz** [higanworks/knife-zero#100]
Looks like a "bootstrap" run on an existing, converged node destroys the local .json file (empty attributes, run_list, roles etc).

Mabye knife-zero can check if a node.json file already exists and prevent doing that?

Besides accidential execution there are legitimate use cases when one wants to update the client.rb file on the node.

**Comment by sawanoboly** [comment]
`knife bootstrap` has --no-converge option. Is it not enough?

**Comment by rmoriz** [comment]
While this option is nice, It should probably be the default to not overwrite/nuke the node.json file. Actually it's so easy to miss and then your node.json is gone. Maybe you can just check if the node file is present and abort? Thank you.

**Comment by sawanoboly** [comment]
Did you mean like https://github.com/higanworks/knife-zero/commit/1f2a88c066f9aad71c8f99b0d2fc710626119043 ?

It asks based on search results from local node objects. like below

```
$ knife zero bootstrap 210.152.xxx.xxx -N knife1
Node "knife1" is already exist. Overwrite it? (y/N) 
```

**Comment by rmoriz** [comment]
yes. that would awesome :-)

**Comment by sawanoboly** [comment]
OK. I'll implement with new option `--[no-]overwrite` false by default.
It will be skiped when `--no-converge` or `--overwrite` option.

**Comment by sawanoboly** [comment]
release as v1.17.0. Could you test it ?

**Comment by rmoriz** [comment]
I'm getting an error when using a (working) FQDN this way:

`knife zero bootstrap working.fqdn.com`

```ruby
ERROR: #<RuntimeError: Error on token '' at 5 of 'name:': Expected expression!>
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/solr/solr_parser.rb:117:in `parse_error'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/solr/solr_parser.rb:124:in `read_single_expression'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/solr/solr_parser.rb:172:in `read_single_expression'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/solr/solr_parser.rb:88:in `read_expression'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/solr/solr_parser.rb:17:in `parse'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/endpoints/search_endpoint.rb:101:in `search'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/endpoints/search_endpoint.rb:14:in `get'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/rest_base.rb:62:in `call'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/rest_router.rb:24:in `call'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/server.rb:664:in `block in app'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/server.rb:336:in `call'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/server.rb:336:in `handle_socketless_request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/socketless_server_map.rb:87:in `request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-zero-4.6.2/lib/chef_zero/socketless_server_map.rb:33:in `request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:305:in `block in send_http_request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:336:in `block in retrying_http_errors'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:334:in `loop'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:334:in `retrying_http_errors'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:299:in `send_http_request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:144:in `request'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/http.rb:111:in `get'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/search/query.rb:158:in `call_rest_service'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/search/query.rb:87:in `search'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/knife-zero-1.17.0/lib/chef/knife/zero_bootstrap.rb:64:in `run'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/knife.rb:421:in `block in run_with_pretty_exceptions'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/local_mode.rb:44:in `with_server_connectivity'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/knife.rb:420:in `run_with_pretty_exceptions'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/knife.rb:219:in `run'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/lib/chef/application/knife.rb:148:in `run'
/Users/rmoriz/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/chef-12.11.18/bin/knife:25:in `<top (required)>'
/Users/rmoriz/.rbenv/versions/2.2.4/bin/knife:23:in `load'
/Users/rmoriz/.rbenv/versions/2.2.4/bin/knife:23:in `<main>'
```

**Comment by sawanoboly** [comment]
Ah... without `-N`. i'd forgotten this case.

I'll fix it.

**Comment by sawanoboly** [comment]
OK. I've released v1.17.1 with fix FQDN only bootstrapping.

I don't like puts foolproof mechanism around for anywhere. I think that you should manage node objects under version control system.
